dallinger/experiment_server/experiment_server.py

```diff
@@ -319,7 +319,10 @@ def advertisement():
         status = part.status
     except exc.SQLAlchemyError:
         status = None
-
+    try:
+        app_id = config.get('id')
```

Comment:
For the second case I think the field should be registered with a default value of None (or some other sentinel value that can be checked to see whether to perform the behavior). Then config.get with that setting name should never raise a KeyError, but return the sentinel value if there was no value loaded.